File: src/main/resources/templates/swift/solution/Sources/swiftExercise/Context.swift
import Foundation

public class Context {
	private var sortAlgorithm: SortStrategy!
	private var dates: [Date]!

	public func getDates() -> [Date] {
		return dates
	}

	public func setDates(_ dates: [Date]) {
		self.dates = dates
	} 

	public func setSortAlgorithm(_ sa: SortStrategy) {
		sortAlgorithm = sa
	}

	public func getSortAlgorithm() -> SortStrategy {
		return sortAlgorithm
	}

	public func sort() {
		sortAlgorithm.performSort(&self.dates)
	}
}

File: src/main/resources/templates/java/maven_blackbox/solution/src/${packageNameFolder}/model/Context.java
package ${packageName}.model;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class Context {
    private SortStrategy sortAlgorithm = new MergeSort();

    private final List<Date> dates = new ArrayList<>();

    public List<Date> getDates() {
        return dates;
    }

    public void addValues(List<Date> values) {
        this.dates.addAll(values);
    }

    public void setSortAlgorithm(SortStrategy sa) {
        sortAlgorithm = sa;
    }

    public SortStrategy getSortAlgorithm() {
        return sortAlgorithm;
    }

    public void clearValues() {
        this.dates.clear();
    }

    /**
     * Runs the configured sort algorithm.
     */
    public void sort() {
        if (sortAlgorithm != null) {
            sortAlgorithm.performSort(this.dates);
        }
    }
}
